src/main/java/com/google/sps/servlets/NewRequestServlet.java

```diff
@@ -0,0 +1,169 @@
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+
+import com.google.firebase.auth.*;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.text.ParseException;
+import java.time.format.DateTimeFormatter; 
+import java.time.format.DateTimeParseException; 
+import java.time.LocalTime;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date; 
+
+@WebServlet("/new-delivery-request")
+public class NewRequestServlet extends HttpServlet {
+  /**
+   * The function handles POST requests sent by delivery-form in deliveryRequest.html 
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    Date deliveryDay = parseDate(request.getParameter("delivery-date"));
+
+    // get the currentDate to check that deliveryDay is not set for a past date
+    Date currentDay = new Date(System.currentTimeMillis());
+    if (deliveryDay == null || deliveryDay.compareTo(currentDay) < 0) {
+      // Send a HTTP 400 Bad Request response if user provided an invalid date.
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Please enter a valid date!");
+      return ;
+    }
+    // TODO[ak47na]: update startTime and endTime with respect to user timezone and check that the
+    // time is not in the past
+    LocalTime startTime = parseTime(request.getParameter("start-time"));
+    LocalTime endTime = parseTime(request.getParameter("end-time"));    
+
+    if (startTime == null || endTime == null || startTime.compareTo(endTime) > 0) {
+      // Send a HTTP 400 Bad Request response if user provided an invalid time.
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Please enter a valid time!");
+      return ;
+    }
+    
+    int maxStops = parseNumber(request.getParameter("max-stops"));
+    if (maxStops <= 0 || maxStops > 25) {
+      // Send a HTTP 400 Bad Request response if user provided an invalid number of stops.
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Please enter between 1 and 25 stops!");
+      return ;
+    }
+    
+    // initialize firebase app with Google Application Defalut Credentials
+    FirebaseApp defaultApp = initializeFirebaseApp();
+    //check the name of defaultApp to make sure that the correct app is connected
+    System.out.println(defaultApp.getName());
+
+    String uid = null;
+    try {
+      // get the idToken from hidden input field
+      String idToken = request.getParameter("idToken");
+      FirebaseToken decodedToken = FirebaseAuth.getInstance().verifyIdToken(idToken);
```

Comment:
Why did you decided to use 'idToken' instead of just doing FirebaseAuth.getCurrentUser() and then get all the info from the user?

Reply:
It was not clear to me how token IDs work, but `FirebaseAuth.getInstance().getCurrentUser() ` is easier and I'll use that in the future.